src/Microsoft.VisualStudio.Editor.Razor/TextBufferFactoryService.cs

```diff
@@ -0,0 +1,16 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using Microsoft.VisualStudio.Text;
+
+namespace Microsoft.VisualStudio.Editor.Razor
+{
+    internal abstract class TextBufferFactoryService
```

Comment:
Chose this name to correspond with some of the other VS/Roslyn types that create/retrieve TextDocument/ITextDocuments. This is the text buffer initializer idea.

Reply:
OK. I'm not crazy about the name but it's internal so #YOLO